src/itensor.jl

```diff
@@ -673,7 +673,9 @@ copy(T::ITensor)::ITensor = itensor(copy(tensor(T)))
 #
 
 # Helper functions for different view behaviors
-Array{ElT,N}(::NeverAlias, A::AbstractArray) where {ElT,N} = Array{ElT,N}(A)
+function Array{ElT,N}(::NeverAlias, A::AbstractArray) where {ElT,N}
+  return NDTensors.similartype(A, ElT)(A)
```

Comment:
I don't think this is right. Constructor calls like `Array{Float64,3}(...)` should be very literal and only output an object with the type `Array{Float64,3}`.